### PR TITLE
BUG: Corrects legend issues other than R2 in PredictionError

### PR DESCRIPTION
**Pull request by pkaf**
This PR addresses the issue raised at #1086

Unfortunately, generic .score() method in scikit-learn does not return the name of the scoring metrics used. The fix I have patched will return correct label if and when estimator has attribute providing the name of the scoring metric otherwise it will fall back to the default value i.e. R2

### CODE REVIEW

**Review by lwgray — APPROVED** [review]
These changes allow for the scoring name to be accurately displayed and default to R2 when scoring name is not provided.  I have tested this with a variety of metrics from sklearn.metrics.SCORERS.keys().  We may need tests to cover name changes.
